ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceAdapter.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection.tasks.multiplechoice
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.ground.R
+import com.google.android.ground.databinding.MultipleChoiceCheckboxItemBinding
+import com.google.android.ground.databinding.MultipleChoiceRadiobuttonItemBinding
+
+/**
+ * An implementation of [ListAdapter] that associates [MultipleChoiceItem]s with their [ViewHolder]
+ */
+class MultipleChoiceAdapter(
+  private val viewModel: MultipleChoiceTaskViewModel,
+  private val canSelectMultiple: Boolean,
+) : ListAdapter<MultipleChoiceItem, RecyclerView.ViewHolder>(MultipleChoiceItemCallback) {
+
+  override fun getItemViewType(position: Int): Int {
+    return when (canSelectMultiple) {
+      true -> R.layout.multiple_choice_checkbox_item
+      else -> R.layout.multiple_choice_radiobutton_item
+    }
```

Comment:
This still doesn't resolve the original question. IMO, we shouldn't reuse the layout ids as item type. It should be some integer constant.

Reply:
Also, I think if / else would be more idiomatic here?

Reply:
Sorry I misunderstood the first comment, fixing!